test/sparse/CS/correctness.chpl

```diff
@@ -32,192 +36,323 @@ proc main() {
   // dsiFirst && dsiLast, for tricky cases
   {
 
-    var csrD: sparse subdomain(D) dmapped csrDmap,
-        cscD: sparse subdomain(D) dmapped cscDmap;
+    var csrsD: sparse subdomain(D) dmapped csrsDmap,
+        cscsD: sparse subdomain(D) dmapped cscsDmap,
+        csruD: sparse subdomain(D) dmapped csruDmap,
+        cscuD: sparse subdomain(D) dmapped cscuDmap;
 
     // Empty domain
-    assert(csrD.first == (0,0));
-    assert(cscD.first == (0,0));
-    assert(csrD.last == (0,0));
-    assert(cscD.last == (0,0));
-
-    csrD += [(1,2), (2,1), (4,5), (5,4)];
-    cscD += [(1,2), (2,1), (4,5), (5,4)];
+    assert(csrsD.first == (0,0));
+    assert(cscsD.first == (0,0));
+    assert(csrsD.last == (0,0));
+    assert(cscsD.last == (0,0));
+    assert(csruD.first == (0,0));
+    assert(cscuD.first == (0,0));
+    assert(csruD.last == (0,0));
+    assert(cscuD.last == (0,0));
```

Comment:
At a glance, it seems like this test would be far more maintainable (and much easier to reason about / ensure correctness for) if it had a generic "test sparse domain" routine that it passed each of the four sparse domains into rather than repeating each operation x the 4 types.  For inspiration, see test/arrays/userAPI/arrayAPItest.chpl and arrayOps2D.chpl (though there's no reason to break it into multiple files necessarily).